core/client/app/components/gh-cm-editor.js

```diff
@@ -0,0 +1,52 @@
+/* global CodeMirror */
+import Ember from 'ember';
+
+var CodeMirrorEditor = Ember.Component.extend({
+
+    // DOM stuff
+    name: 'cm-editor', // this is so there is a value, just in case an id isn't specified
+    classNameBindings: ['isFocused:focused'],
+    isFocused: false,
+
+    value: '', // make sure a value exists
+    editor: null, // reference to CodeMirror editor
+    mode: 'htmlmixed',
+    theme: 'base16-light',
+
+    // list of default options
+    options: {
+        lineNumbers: true,
+        indentUnit: 4
+    },
+
+    didInsertElement: function () {
+        var options = this.get('options'),
+            self = this,
+            editor;
+        options.mode = this.get('mode');
+        options.theme = this.get('theme');
```

Comment:
You can get at your options a bit more quickly like so, if you like this style:
```js
// codemirror options
mode: 'foo',
theme: 'bar',
lineNumbers: true,
indentUnit: 4,
//...
didInsertElement () {
  var options = this.getProperties('mode', 'theme', 'lineNumbers', 'indentUnit');
//  ....
}
```
not necessary, I just thought it was odd to half-build options outside the function, then add more options inside of the function :)